### PR TITLE
Unbreak the E2E tests

### LINKED ISSUE #361: Nightly releases are published even if E2E tests fail
e.g. https://storage.googleapis.com/knative-prow/logs/ci-knative-build-pipeline-nightly-release/1080018717502345217/build-log.txt

E2E tests are not propagating the return code correctly. In the log above, the sequence is:

```
I0101 08:58:21.606] FAIL	github.com/knative/build-pipeline/test	1107.544s
I0101 08:58:21.618] Finished run, return code is 1
...
I0101 08:58:47.592] =============================================
I0101 08:58:47.592] ==== RUNNING YAML E2E TESTS FOR TASKRUNS ====
I0101 08:58:47.593] =============================================
...
I0101 08:58:53.679] **************************************
I0101 08:58:53.679] ***        E2E TESTS PASSED        ***
I0101 08:58:53.679] **************************************
I0101 08:58:53.681] =======================================
I0101 08:58:53.682] ==== TEARING DOWN TEST ENVIRONMENT ====
I0101 08:58:53.682] =======================================
I0101 08:58:53.824] ***************************************
I0101 08:58:53.824] ***         E2E TEST FAILED         ***
I0101 08:58:53.824] ***    Start of information dump    ***
I0101 08:58:53.825] ***************************************
I0101 08:58:53.825] >>> All resources:
...
I0101 09:04:21.184] Test subprocess exited with code 1
I0101 09:04:22.923] Test result code is 0
```

go tests failed, YAML tests passed. The teardown correcty shows that tests failed, but the test result is 0. Looks like the YAML tests overrode the test result.

### LINKED ISSUE #368: E2E tests broken at head due to broken rename in #366
## Expected Behavior
E2E tests should pass at head.

## Actual Behavior
E2E tests fail at head (e.g. in #367) due to changes in #366 - and even in that PR you can see that the integration tests actually failed with:

```
I0104 12:48:05.037] Running tests with 'go test -race -v  -count=1 -tags=e2e -timeout=20m ./test '
W0104 12:48:45.938] # github.com/knative/build-pipeline/test
W0104 12:48:45.938] test/helm_task_test.go:277:19: unknown field 'PipelineTrigger' in struct literal of type "github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1".PipelineRunSpec
W0104 12:48:45.939] test/pipelinerun_test.go:466:19: unknown field 'PipelineTrigger' in struct literal of type "github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1".PipelineRunSpec
W0104 12:48:45.939] test/pipelinerun_test.go:593:19: unknown field 'PipelineTrigger' in struct literal of type "github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1".PipelineRunSpec
I0104 12:48:46.040] FAIL	github.com/knative/build-pipeline/test [build failed]
```

Though the github hook reported success (maybe related to #361?). As of #367 the pre merge hook seems to actually be reporting failure correctly - OR something else is failing and causing the hook to report failure?

### PR DESCRIPTION
**Pull request by adrcunha**
Fixes #361.
Fixes #368.

## Update test-infra

Major changes:
* Also update E2E test return value on failures
* Default build/unit test runners

## Fix botched rename of PipelineTrigger field

This is a copy of #369.

## Disable broken tests

There are 2 tests failing consistently, and it seems that they're failing only on Prow. Disabling them here to unblock the E2E tests, filed #375 and #376 to track the fixes.

### CODE REVIEW

File: test/presubmit-tests.sh [hunk 1]
```diff
@@ -23,26 +23,9 @@
 
 source $(dirname $0)/../vendor/github.com/knative/test-infra/scripts/presubmit-tests.sh
 
-function build_tests() {
-  header "Verifying code structure and contents"
+# We use the default build test runner.
```
Comment (ImJasonH) [inline test/presubmit-tests.sh]: This line is duplicated (mostly)

Comment (adrcunha) [reply test/presubmit-tests.sh]: No, it's not. Each one refers to a different runner: build, unit and integration.

**Comment by adrcunha** [comment]
Nevermind the retests, this is not flakiness:

```
I0109 15:14:22.583]     --- FAIL: TestPipelineRun/fan-in_and_fan-out (600.37s)
I0109 15:14:22.583]         pipelinerun_test.go:153: Error waiting for PipelineRun helloworld-pipelinerun1 to finish: timed out waiting for the condition
...
I0109 15:16:14.195] error: error validating "STDIN": error validating data: ValidationError(ClusterRoleBinding): unknown field "spec" in io.k8s.api.rbac.v1beta1.ClusterRoleBinding; if you choose to ignore these errors, turn validation off with --validate=false
I0109 15:16:14.200] 2019/01/09 15:16:14 error executing 'kubectl apply': exit status 1
I0109 15:16:14.201] ERROR: one or more YAML tests failed
```

@bobcatfish do you know what's wrong here? I don't think that this is a simple case of increasing the timeout (and it won't fix the yaml tests anyway)

**Comment by adrcunha** [comment]
As discussed on Slack, the errors in E2E seem to be related to the Prow environment. I'm gonna disable the broken tests to unblock the pipeline and file issues for them.

**Comment by adrcunha** [comment]
Tests passed, let me file issues and update the PR description...

/hold

**Comment by adrcunha** [comment]
Filed #375 and #376 for the broken tests.

**Comment by ImJasonH** [comment]
/lgtm
/approve

:tada:

**Pull request merged**